build/dockerfiles/Dockerfile.rhel

```diff
@@ -41,9 +42,10 @@ RUN chmod a+rwX            /run/httpd
 RUN mkdir /var/www/html/devfiles
 COPY .htaccess README.md /var/www/html/
 COPY --from=builder /build/devfiles /var/www/html/devfiles
-USER 0
-RUN chmod -R g+rwX /var/www/html/devfiles
 
 STOPSIGNAL SIGWINCH
 
-ENTRYPOINT ["/usr/sbin/httpd", "-D", "FOREGROUND"]
+COPY --from=builder /build/devfiles /var/www/html/devfiles
+COPY ./build/dockerfiles/entrypoint.sh /usr/bin/
+ENTRYPOINT ["/usr/bin/entrypoint.sh"]
+CMD ["/usr/sbin/httpd", "-D", "FOREGROUND"]
```

Comment:
we could put this INSIDE the entrypoint.sh script. What benefit to being separate? Also you can have the same entrypoint check for the alpine/ubuntu vs. rhel path to run httpd:

```
# start httpd
if [[ -x /usr/sbin/httpd ]]; then /usr/sbin/httpd -D FOREGROUND
elif [[ -x /usr/bin/run-httpd ]]; then /usr/bin/run-httpd
fi
```

(You might recognize that from https://github.com/eclipse/che-plugin-registry/blob/master/scripts/entrypoint.sh )

Reply:
The benefit of having the `httpd` command as a CMD instead of in the ENTRYPOINT is that it's easier to override for debugging:
```
docker run -it --rm <registry> /bin/bash
```

This is generally how it's done AFAIK, e.g. for `registry.access.redhat.com/rhscl/httpd-24-rhel7:2.4-104` we have
```
"Entrypoint": ["container-entrypoint"],
"Cmd": ["/usr/bin/run-httpd"],
```

As for the actual command being run, I'm copying what was done previously (as an entrypoint instead of a command). I can add the wrapper script from plugin registry if necessary, though.

Reply:
Added `rhel.entrypoint.sh` script to match plugin registry structure.

Reply:
but you can:

`docker run -it --rm --entrypoint /bin/bash <container> -c "ls -la ${HOME}"` too, regardless of the stated ENTRYPOINT or CMD, right?

(key is to change the order of the command and explicitly use `--entrypoint /bin/bash` BEFORE the container `reg/org/image:tag`)

Reply:
Yeah but that's not the typical way docker images are written -- there's a reason you need an additional flag. In fact, that's not the way other centos httpd container implementations are written.

Also, your form makes it harder to test that the entrypoint is doing what we expect -- to test the container you would have to 1) override entrypoint and start with `/bin/bash`, 2) run the entrypoint from bash.